### PR TITLE
dataflow/custom-containers: add readmes and support cli options

### PR DESCRIPTION
**Pull request by davidcavazos**
## Description

This makes the following changes to the Dataflow custom container samples:
* Adds a README file with instructions on how to run
* Supports command line arguments on the pipelines
* Update to Python 3.9
* Update to Beam 2.37

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.6` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

**Review by dandhlee — CHANGES_REQUESTED** [review]

File: dataflow/custom-containers/miniconda/e2e_test.py [hunk 1]
```diff
@@ -37,18 +37,21 @@ def container_image(utils: Utils) -> str:
     yield from utils.cloud_build_submit(image_name=NAME)
 
 
-def test_tensorflow_minimal(
-    utils: Utils, bucket_name: str, container_image: str
-) -> None:
-    subprocess.check_call(
```
[the remark this answers is not in the document]
Comment (davidcavazos) [reply dataflow/custom-containers/miniconda/e2e_test.py:43]: Thanks, removed!

File: dataflow/custom-containers/miniconda/e2e_test.py [hunk 2]
```diff
@@ -37,18 +37,21 @@ def container_image(utils: Utils) -> str:
...
-            f"--temp_location=gs://{bucket_name}",
-            f"--sdk_container_image={container_image}",
-            "--experiment=use_runner_v2",
-        ]
+@pytest.fixture(scope="session")
+def run_dataflow_job(utils: Utils, bucket_name: str, container_image: str) -> str:
+    yield from utils.cloud_build_submit(
+        config="run.yaml",
+        substitutions={
+            "_JOB_NAME": utils.hyphen_name(NAME),
```
[the remark this answers is not in the document]
Comment (davidcavazos) [reply dataflow/custom-containers/miniconda/e2e_test.py:43]: Adding to requirements-test.txt

**Comment by dandhlee** [comment]
Getting the following error now:
```
Traceback (most recent call last):
  File "/workspace/dataflow/custom-containers/miniconda/e2e_test.py", line 54, in test_tensorflow_minimal
    job_id = utils.dataflow_job_id(utils.hyphen_name(NAME))
  File "/workspace/dataflow/conftest.py", line 339, in dataflow_job_id
    raise ValueError(f"Dataflow job not found: job_name={job_name}")
ValueError: Dataflow job not found: job_name=dataflow-custom-containers-miniconda-py39-c49859
```

**Review by dandhlee — APPROVED** [review]
Yay! LGTM.

**Review by engelke — APPROVED** [review]

**Pull request merged**